app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  validates :name, presence: true,
+  length: {maximum: Settings.user.validate.name_max}
+  VALID_EMAIL_REGEX = Settings.user.validate.VALID_EMAIL_REGEX
+  validates :email, presence: true,
+  length: {maximum: Settings.user.validate.email_max},
+  format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  validates :password, presence: true,
+  length: {minimum: Settings.user.validate.pass_min}
+
+  has_secure_password
```

Comment:
cái này để làm gì bạn ?

Reply:
has_secure_password được dùng để mã hóa và xác nhận bằng cách sử dụng mật khẩu bcrypt. Lưu thuộc tính password_digest được băm an toàn vào cơ sở dữ liệu. Xác thực hiện diện của password, nó  áp dụng cho các bản ghi có mật khẩu trống, cho phép người dùng tạo mật khẩu không hợp lệ như ''      "(6 dấu cách ) và yêu cầu password và password-confirmation phải khớp . has_secure_password tự động thêm một phương thức xác thực vào các đối tượng tương ứng .Phương pháp này xác định xem một mật khẩu đã cho có hợp lệ đối với một người dùng cụ thể hay không bằng cách tính toán và so sánh kết quả với password_digest trong cơ sở dữ liệu.